### ISSUE TITLE
Numerical Answer with Exponent Does Not Parse

### ISSUE THREAD

**Issue by jIyajbe** [maths/moodle-qtype_stack#90]
Hi,

I'm having a problem creating a STACK question that uses exponential notation. My version of STACK is 3.1 (2014011500), my version of Maxima is 5.30.0, and my OS is OSX 10.75 ("Lion").

I've created the following STACK question:

**Question Variables:
E : rand(500.0) + 0.5;
L: rand(20.0)+1.25;
c: 299792458.0;
ta1: float(c/L);
ta2: z;
ta3: float(E/c);
ta4: float((2_%pi)/L);
ta5: float(2_%pi*c/L);**

Question Text:
A plane electromagnetic wave, with wavelength @L@ m, travels in free space in the +x-direction with its electric vector E, of amplitude @E@ V/m, directed along the y-axis.

(a) What is the frequency of the wave?

ν = [[input:ans1]] Hz
[[validation:ans1]](b) What is the direction and amplitude of the magnetic field associated with the wave?

direction = [[input:ans2]](For the direction, enter either x, y, z, or the word "other" [without the quotes]. If the B-field is pointing in a negative direction, include the negative sign in your answer)
[[validation:ans2]]

Bm = [[input:ans3]] T
[[validation:ans3]](c) If E = Emsin(kx - ωt), what are the values of k and ω?

k = [[input:ans4]]  m-1
[[validation:ans4]]

ω = [[input:ans5]]  s-1
[[validation:ans5]]

Next, I _Preview_ the question, and obtain (for example) the following values:
L = 11.53380994
E = 181.984425412

I click "Fill in correct responses", and get the following:

--For ans1 (the frequency), I get an error:

**"Your last answer was interpreted as follows: 2.599249160155e+7
This answer is invalid.
You seem to be missing \* characters. Perhaps you meant to type 2.599249160155*e+7."**

NOTE: I get similar errors for Bm and for ω.

--For ans4 (the wavenumber k), it succeeds: 
**"Your last answer was interpreted as follows: .544762341312 "**

In short:
**Every numeric value that evaluates to engineering (exponential) notation fails as above, and every numeric value that evaluates to a simple float succeeds.**
